flocker/control/_persistence.py

```diff
@@ -91,6 +169,10 @@ def decode_object(dictionary):
                            [_DEPLOYMENT_FIELD])
 _LOG_SAVE = ActionType(u"flocker-control:persistence:save",
                        [_DEPLOYMENT_FIELD], [])
+_LOG_UPGRADE = ActionType(u"flocker-control:persistence:migrate_configuration",
+                          [_DEPLOYMENT_FIELD,
+                           Field(u"source_version", repr),
+                           Field(u"target_version", repr)], [])
```

Comment:
I think these should instead be something like `Field.for_types(u"source_version", [int], ...)`.  Also fields should always have documentation strings (3rd argument to `for_types`).